### PR TITLE
Create Package

### PR DESCRIPTION
**Pull request by mem48**
Some basic package structure and a function to return SF objects from routes

### CODE REVIEW

**Review by marcusyoung — COMMENTED** [review]
Thanks @mem48 for this. A few thoughts:

1. We should be consistent with function naming. I had adopted a prefix of lowercase otp followed by function name in camel case. This should avoid many (any?) clashes with other packages. So PlanRoute would become otpPlanRoute. Do you think this is reasonable? Other ideas?

2. I had proposed the otpConnect function to define the base url rather than  having it specified in full in the function call. At the moment otpConnect  generates the base url from supplied arguments, but my intention was that this would do some querying of the OTP instance, confirming that it was live and perhaps returning some potentially useful information, such as a list of routers on that instance. When I was working with OTP I had multiple routers running on the same instance as well as multiple instances and it quickly became problematic having to specify the base url in each function call, and was prone to error.

3. Is PlanRoute intended to be an additional function to exist alongside the other functions such as  otpDistance and otpTime or a replacement? These simpler functions become quite important when performing hundreds of thousands of queries because they minimise the network overhead involved in returning the data (in many cases the OTP server will not co-exist with the R client and may be cloud-hosted) and the R processing overhead.

File: R/json2sf.R [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#' Convert Google Encoded Polyline into sf object
+#'
+#' @param line character - polyline
+#' @export
+#' @examples
+#' None
+polyline2linestring <- function(line){
+  line = gepaf::decodePolyline(line)
```
Comment (Robinlovelace) [inline R/json2sf.R:8]: @marcusyoung and @mem48 I suggest using this package instead: a more efficient and self-contained implementation I think: https://github.com/SymbolixAU/googlePolylines

**Comment by mem48** [comment]
@marcusyoung 

To address your points
1. Yes you are right
2. Ok, I removed the function because it didn't seem very useful, however now you've explained it seems more necessary
3. This I think is a stylistic decision, we could have one mega function with parameters that dictate what you pull back, or many similar functions that pre-select certain variables. I'm not sure it will save any bandwidth because your functions retrieve the same json file mine does, but I've put more code into transforming and processing the json which will add processing time at the R side. 

Perhaps we need a outline list of functions and then we can decide what goes where.

**Comment by marcusyoung** [comment]
@mem48 

You may have seen that I've got a pull request for the otp_connect function to get feedback. I don't think you've accepted the invite to be a collaborator, so I've not been able to invite you to review?

I've now put a list of proposed functions in the issues. As you'll see I've suggested separate functions to perform specific tasks. I wonder if that will be easier for users to understand and implement then a mega function where there would be a substantially greater initial learning curve? This approach may also be more manageable from a development and bug-fixing point of view? But open to discussion on this.

Take your point on the bandwidth, as the same json file is being pulled.

**Pull request closed without merging**